packages/frontend/src/components/Header.jsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react'
+import React, { useContext, useState,useEffect } from 'react'
 import { Outlet, Link } from "react-router-dom";
+import { observer } from 'mobx-react-lite'
+import state from '../contexts/state';
 import './header.css'
 
-export default () => {
+export default observer(() => {
+    const { unirep } = useContext(state)
+    const [userIds, setUserIds] = useState([])
+    const [attesterIds, setAttesterIds] = useState([])
+    useEffect(() => {
+    const loadData = async () => {
+        await unirep.loadAllSignUps()
+        await unirep.loadAllAttestations()
+        await unirep.loadAllEpochs()
```

Comment:
These calls are happening one after another right now, instead you want to do them in parallel to get the data as quickly as possible.

e.g.

```js
await Promise.all([
  unirep.loadAllSignUps(),
  unirep.loadAllAttestations(),
  unirep.loadAllEpochs()
])